atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTopologyValidatorTest.java

```diff
@@ -0,0 +1,304 @@
+        Map<CassandraServer, CassandraClientPoolingContainer> allHosts = setupHosts(NEW_HOSTS);
+        EntryStream.of(allHosts)
+                .values()
+                .forEach(container -> setHostIds(
+                        Set.of(container),
+                        HostIdResult.success(Set.of(uuidIterator.next())),
+                        HostIdResult.success(UUIDS)));
+        assertThat(validator.getNewHostsWithInconsistentTopologiesAndRetry(
+                        allHosts.keySet(),
+                        allHosts,
+                        HumanReadableDuration.milliseconds(1),
+                        HumanReadableDuration.seconds(20)))
+                .isEmpty();
+    }
+
+    @Test
+    public void retriesThrowsAndMarksFailureWhenNeverResolves() {
+        Iterator<String> uuidIterator = UUIDS.iterator();
+        Map<CassandraServer, CassandraClientPoolingContainer> allHosts = setupHosts(NEW_HOSTS);
+        EntryStream.of(allHosts)
+                .values()
+                .forEach(container -> setHostIds(Set.of(container), HostIdResult.success(Set.of(uuidIterator.next()))));
+        assertThat(validator.getNewHostsWithInconsistentTopologiesAndRetry(
+                        allHosts.keySet(),
+                        allHosts,
+                        HumanReadableDuration.milliseconds(1),
+                        HumanReadableDuration.milliseconds(1)))
+                .isNotEmpty();
+        assertThat(metrics.validationFailures().getCount()).isEqualTo(1);
+        assertThat(metrics.validationLatency().getCount()).isEqualTo(1);
+    }
+
+    @Test
+    public void returnsEmptyWhenGivenNoServers() {
+        assertThat(validator.getNewHostsWithInconsistentTopologies(ImmutableSet.of(), ImmutableMap.of()))
+                .isEmpty();
+    }
+
+    @Test
+    public void throwsWhenAllHostsDoesNotContainNewHosts() {
+        assertThatThrownBy(() -> validator.getNewHostsWithInconsistentTopologies(
+                        ImmutableSet.of(createCassandraServer("foo")), ImmutableMap.of()))
+                .isInstanceOf(SafeIllegalArgumentException.class);
+    }
+
+    @Test
+    public void returnsEmptyWhenOnlyNewServersAndAllHaveSameHostIds() {
+        Map<CassandraServer, CassandraClientPoolingContainer> allHosts = setupHosts(NEW_HOSTS);
+        setHostIds(allHosts.values(), DEFAULT_RESULT);
+        assertThat(validator.getNewHostsWithInconsistentTopologies(allHosts.keySet(), allHosts))
+                .isEmpty();
+    }
+
+    @Test
+    public void returnsNewHostsWhenOnlyNewServersAndOneMismatch() {
+        Map<CassandraServer, CassandraClientPoolingContainer> allHosts = setupHosts(NEW_HOSTS);
+        setHostIds(
+                filterContainers(allHosts, server -> !server.equals(NEW_HOST_ONE)),
+                HostIdResult.success(ImmutableSet.of("uuid1", "uuid2")));
+        setHostIds(
+                filterContainers(allHosts, NEW_HOST_ONE::equalsIgnoreCase),
```

Comment:
question: I'm guessing using this (and not just `equals()`) is some artifact of how Thrift works? 😬

Reply:
I'll change them to `equals`, but in this case I did it b/c it makes writing tests that much easier and less cumbersome if some decides to change type case of the server names!